test/libsolidity/smtCheckerTests/complex/warn_on_struct.sol

```diff
@@ -1,15 +1,11 @@
 pragma experimental SMTChecker;
 
 contract C {
-    struct A { uint a; uint b; }
-    function f() public pure returns (uint) {
-        A memory a = A({ a: 1, b: 2 });
-    }
+	struct A { uint a; uint b; }
+	function f() public pure returns (uint) {
+		A memory a = A({ a: 1, b: 2 });
+		assert(a.a == 1 && a.b == 2);
+	}
 }
 // ----
-// Warning 6321: (117-121): Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
-// Warning 2072: (133-143): Unused local variable.
-// Warning 8364: (146-147): Assertion checker does not yet implement type type(struct C.A storage pointer)
-// Warning 4639: (146-163): Assertion checker does not yet implement this expression.
-// Warning 8364: (146-147): Assertion checker does not yet implement type type(struct C.A storage pointer)
-// Warning 4639: (146-163): Assertion checker does not yet implement this expression.
+// Warning 6321: (111-115): Unnamed return variable can remain unassigned. Add an explicit return with value to all non-reverting code paths or name the variable.
```

Comment:
Since the following test also passes, it could be added

```
pragma experimental SMTChecker;

contract C {
        struct B { uint b1; }
        struct A { uint a1; B a2; }
        function f() public pure {
                A memory a = A({ a1: 1, a2: B({b1: 2}) });
                assert(a.a1 == 1 && a.a2.b1 == 2);
        }
}
```

Reply:
If we change the function to not return anything (and thus remove the Warning 6321) then we can move this test to the `types/struct` where the rest of related tests is located. Maybe it can even be deleted, because it will be subsumed by the tests that are already there.
@leonardoalt , what do you think?

Reply:
If we already have tests there that also contain construction of nested structs I'd say we can delete

Reply:
I am adding the example suggested by @bshastry and there are already tests there for simple struct with multiple fields, which is the case of `complex/warn_on_struct.sol`